index.js

```diff
@@ -17,7 +17,7 @@ module.exports = function(options) {
             return cb(new gutil.PluginError('gulp-image-optimization', 'Streaming not supported'));
         }
 
-        if (['.jpg', '.jpeg', '.png', '.gif'].indexOf(path.extname(file.path)) === -1) {
+        if (['.jpg', '.JPG', '.JPEG', '.jpeg', '.png', '.PNG', '.gif', '.GIF'].indexOf(path.extname(file.path)) === -1) {
```

Comment:
Yes, on case-sensitive file systems (like most Linux filesystems).

Alternatively the code could be written as .indexOf(path.extname(file.path.toLowerCase()) and then just have lower case extensions, but as (I presume) we don't want to match paths ending in eg ".JpEg" then I thought a list of capitalised extensions was a simpler change.